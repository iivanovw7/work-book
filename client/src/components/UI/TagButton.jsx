import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { ripples } from '../../styles';
/* eslint react/require-default-props: 0 */

const StyledWrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	flex-wrap: wrap;
	font-size: 0.8em;
	outline: none;
`;

const Button = styled.button`
	${ripples};
  float: none;
  display: inline-block;
  border-radius: 0.2em;
  border: 0;
  outline: none;
  margin: ${props => (props.margin ? props.margin : '0.2em')};
  background-color: ${props => props.bgColor};
  color: ${props => props.textColor};
  text-decoration: none;
  
  &:hover {
    user-select: none;
    cursor: pointer;
    outline: none;
    border: 0;
  }
  
  &:focus {
    outline: none;
    border: 0;
  }
  
  &::-moz-focus-inner {
    border: 0;
  }

`;

const TagButton = (props) => {
  const {
    handleClick, text, bgColor, textColor, theme, label, margin
  } = props;

  return (
    <ThemeProvider theme={{ mode: theme }}>
      <StyledWrapper className="dates">
        {label}
        <Button
          margin={margin}
          onClick={handleClick}
          bgColor={bgColor}
          textColor={textColor}
          {...props}
        >
          {text}
        </Button>
      </StyledWrapper>
    </ThemeProvider>
  );
};

TagButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  label: PropTypes.string,
  bgColor: PropTypes.string,
  textColor: PropTypes.string,
  margin: PropTypes.string
};

export default TagButton;
