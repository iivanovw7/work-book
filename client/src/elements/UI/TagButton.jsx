import PropTypes from 'prop-types';
import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import { ripples } from '../../styles';

const StyledWrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	flex-wrap: wrap;
	font-size: 0.8em;
`;

const Button = styled.button`
	${ripples};
  float: none;
  display: inline-block;
  border-radius: 0.2em;
  border: 0;
  outline: 0;
  margin: 0.2em;
  background-color: ${props => props.bgColor};
  color: ${props => props.textColor}
  
  &:hover {
    user-select: none;
    cursor: pointer;
  }
`;

const TagButton = (props) => {
	const {
		handleClick, text, bgColor, textColor, theme, label
	} = props;

	return (
		<ThemeProvider theme={{ mode: theme }}>
			<StyledWrapper className="dates">
				{label}
				<Button
					onClick={handleClick}
					bgColor={bgColor}
					textColor={textColor}
					{...props}
				>
					{text}
				</Button>
			</StyledWrapper>
		</ThemeProvider>
	);
};

TagButton.propTypes = {
	handleClick: PropTypes.func.isRequired,
	text: PropTypes.string.isRequired,
	theme: PropTypes.string.isRequired,
	label: PropTypes.string,
	bgColor: PropTypes.string,
	textColor: PropTypes.string
};

export default TagButton;
