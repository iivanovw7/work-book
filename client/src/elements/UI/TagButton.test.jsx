import 'cross-fetch/polyfill';
import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';
import TagButton from './TagButton';

jest.mock('../../config/apiURL', () => () => '/api');

describe('Testing TagButton component', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  const props = {
    text: 'text',
    theme: 'dark',
    label: 'label',
    bgColor: 'bg',
    textColor: 'color'
  };

  it('Should match snapshot', () => {
    const mockFunc = jest.fn();

    // Disabling PropTypes console.log warning until component is rendered
    const originalError = console.error;
    console.error = jest.fn();
    const component = mount(<TagButton handleClick={mockFunc()} {...props} />);
    console.error = originalError;

    expect(component).toMatchSnapshot();
  });

  it('Should render with correct props', () => {
    const mockFunc = jest.fn();

    // Disabling PropTypes console.log warning until component is rendered
    const originalError = console.error;
    console.error = jest.fn();
    const component = mount(<TagButton handleClick={mockFunc()} {...props} />);
    console.error = originalError;

    expect(component.find(TagButton).prop('textColor')).toBe('color');
    expect(component.find(TagButton).prop('bgColor')).toBe('bg');
  });

  it('Should trigger click event', () => {
    const mockFunc = jest.fn();

    // Disabling PropTypes console.log warning until component is rendered
    const originalError = console.error;
    console.error = jest.fn();
    const component = mount(<TagButton handleClick={mockFunc()} {...props} />);
    console.error = originalError;

    expect(component.find(TagButton)).toHaveLength(1);
    component.find(TagButton).at(0).simulate('click');
    expect(mockFunc).toHaveBeenCalled();
  });
});
