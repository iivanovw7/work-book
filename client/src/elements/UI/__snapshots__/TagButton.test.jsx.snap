// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TagButton component Should match snapshot 1`] = `ReactWrapper {}`;
